services/data/src/react/hooks/stableVariablesHash.ts

```diff
@@ -0,0 +1,58 @@
+function hasObjectPrototype(o: any): boolean {
+    return Object.prototype.toString.call(o) === '[object Object]'
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/explicit-module-boundary-types
+export function isPlainObject(o: any): o is Object {
+    if (!hasObjectPrototype(o)) {
+        return false
+    }
+
+    // If has modified constructor
+    const ctor = o.constructor
+    if (typeof ctor === 'undefined') {
+        return true
+    }
+
+    // If has modified prototype
+    const prot = ctor.prototype
+    if (!hasObjectPrototype(prot)) {
+        return false
+    }
```

Comment:
Throws if `prototype` is `undefined`

Reply:
`isPlainObject(Object.create({}))` imo should return true but `Object.create({}).prototype === undefined`

Reply:
This was taken from react-query's implementation, see [here](https://github.com/tannerlinsley/react-query/blob/76e314d0887edbd702369e8554cc840508561b94/src/core/utils.ts#L392) and above the code from the link.

So I'm fine with this failing and more concerned with staying close to react-query (as we want the results to be similar). If we want to fix this we should fix this upstream as otherwise our code won't fail but react-query still would.